Initium-Odp/src/com/universeprojects/miniup/server/controllers/ViewCharacterMiniController.java

```diff
@@ -0,0 +1,219 @@
+package com.universeprojects.miniup.server.controllers;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Collections;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.Key;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.universeprojects.cacheddatastore.ShardedCounterService;
+import com.universeprojects.miniup.CommonChecks;
+import com.universeprojects.miniup.server.GameUtils;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.web.Controller;
+import com.universeprojects.web.PageController;
+
+/**
+ * Controller for the view_profile page.
+ * Available request params:
+ *
+**/
+
+@Controller
+public class ViewCharacterMiniController extends PageController {
+
+	public ViewCharacterMiniController() {
+		super("viewcharactermini");
+	}
+	
+	@Override
+	protected String processRequest(HttpServletRequest request,
+			HttpServletResponse response) throws ServletException, IOException {
+		response.setHeader("Access-Control-Allow-Origin", "*");     // This is absolutely necessary for phonegap to work
+		
+		ODPDBAccess db = ODPDBAccess.getInstance(request);
+		CachedEntity character = db.getCurrentCharacter();
```

Comment:
`character` should be the character we're trying to view. Look in `ViewItemController` to see how we get the `characterId` attribute from the request to get a character entity.

```Long charId = WebUtils.getLongParam(request, "characterId");
if (charId==null) return null;
Key characterKey = KeyFactory.createKey("Character", charId);
CachedEntity character = db.getEntity(characterKey); ```

Don't forget to validate that the character exists after retrieving.